flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/windowing/FoldApplyProcessWindowFunction.java

```diff
@@ -92,12 +93,45 @@ public void process(K key, final Context context, Iterable<T> values, Collector<
 			result = foldFunction.fold(result, val);
 		}
 
-		windowFunction.process(key, windowFunction.new Context() {
+		ProcessWindowFunction<ACC, R, K, W>.Context ctx = windowFunction.new Context() {
```

Comment:
This can benefit from a similar refactoring as the internal window functions, i.e. creating an internal context class instead of the anonymous inner classes.

This also holds for `FoldApplyProcessAllWindowFunction`, `ReduceApplyProcessAllWindowFunction` and `ReduceApplyProcessWindowFunction`.